Jellyfin.Server.Implementations/AudioWaveform/AudioWaveformManager.cs

```diff
@@ -0,0 +1,170 @@
+    /// <param name="fileSystem">The file system.</param>
+    /// <param name="encodingHelper">The encoding helper.</param>
+    /// <param name="libraryManager">The library manager.</param>
+    /// <param name="config">The server configuration manager.</param>
+    /// <param name="imageEncoder">The image encoder.</param>
+    /// <param name="dbProvider">The database provider.</param>
+    /// <param name="appPaths">The application paths.</param>
+    public AudioWaveformManager(
+        ILogger<AudioWaveformManager> logger,
+        IMediaEncoder mediaEncoder,
+        IFileSystem fileSystem,
+        EncodingHelper encodingHelper,
+        ILibraryManager libraryManager,
+        IServerConfigurationManager config,
+        IImageEncoder imageEncoder,
+        IDbContextFactory<JellyfinDbContext> dbProvider,
+        IApplicationPaths appPaths)
+    {
+        _logger = logger;
+        _mediaEncoder = mediaEncoder;
+        _fileSystem = fileSystem;
+        _encodingHelper = encodingHelper;
+        _libraryManager = libraryManager;
+        _dbProvider = dbProvider;
+        _appPaths = appPaths;
+    }
+
+    /// <inheritdoc />
+    public async Task<FileStream> GetAudioWaveformAnsyc(Guid itemId)
+    {
+        var tempDir = Path.Combine(_appPaths.TempDirectory, "audiowaveform_" + itemId);
+        Directory.CreateDirectory(tempDir);
+
+        var item = _libraryManager.GetItemById<BaseItem>(itemId)
+            ?? throw new ResourceNotFoundException();
+
+        var libraryOptions = BaseItem.LibraryManager.GetLibraryOptions(item);
+        var saveInMediaFolder = libraryOptions.SaveLocalMetadata;
+        var saveFileName = Path.GetFileNameWithoutExtension(item.Path) + ".json";
+        string outputPathJson = saveInMediaFolder
+            ? Path.Combine(item.ContainingFolderPath, saveFileName)
+            : Path.Combine(item.GetInternalMetadataPath(), saveFileName);
+        string outputPathCsv = Path.Combine(tempDir, saveFileName.Replace("json", "csv", StringComparison.Ordinal));
+
+        if (!File.Exists(outputPathJson))
+        {
+            int sampleRate = 0;
+            int fileSampleRate = item.GetMediaStreams()[0].SampleRate ?? 44100;
+            // This value sets the length of a frame, the amount of measurements per second.
+            int samplesPerSecond = 2;
+            sampleRate = Math.Max(fileSampleRate / samplesPerSecond, 1);
+
+            // escape double quotes
+            string itemPathEscaped = item.Path.Replace("\"", "\\\"", StringComparison.Ordinal);
+            string outputPathCsvEscaped = outputPathCsv.Replace("\"", "\\\"", StringComparison.Ordinal);
+
+            var process = Process.Start("ffprobe", $"-v error -f lavfi -i \"amovie={itemPathEscaped},asetnsamples={sampleRate},astats=metadata=1:reset=1\" " + $"-show_entries frame_tags=lavfi.astats.Overall.RMS_peak -of csv=p=0 -o \"{outputPathCsvEscaped}\"");
+
+            var waitTask = process.WaitForExitAsync();
```

Comment:
```suggestion
            await process.WaitForExitAsync().WaitAsync(Timespan.FromSecounds(10));
```

https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.waitasync?view=net-9.0#system-threading-tasks-task-waitasync(system-timespan)